src/ui/pages/managers/createManager.page.ts

```diff
@@ -0,0 +1,259 @@
+    await this.clearAllButton.click();
+  }
+
+  @logStep('Click back link')
+  async clickBackLink() {
+    await this.backLink.click();
+  }
+
+  @logStep('Fill complete manager form')
+  async fillManagerForm(managerData: ICreateManagerData) {
+    await this.fillUsername(managerData.username);
+    await this.fillFirstName(managerData.firstName);
+    await this.fillLastName(managerData.lastName);
+    await this.fillPassword(managerData.password);
+    await this.fillConfirmPassword(managerData.confirmPassword);
+  }
+
+  @logStep('Create manager with data')
+  async createManager(managerData: ICreateManagerData) {
+    await this.fillManagerForm(managerData);
+    await this.clickSaveButton();
+  }
+
+  // Validation methods
+  @logStep('Check field validation state: {fieldName}')
+  async checkFieldValidationState(fieldName: string, expectedState: 'valid' | 'invalid') {
+    const field = this.getFieldByName(fieldName);
+
+    if (expectedState === 'valid') {
+      await expect(field).toHaveClass(/is-valid/);
+    } else {
+      await expect(field).toHaveClass(/is-invalid/);
+    }
+  }
+
+  @logStep('Check field validation color: {fieldName}')
+  async checkFieldValidationColor(fieldName: string, expectedColor: 'green' | 'red') {
+    const field = this.getFieldByName(fieldName);
+
+    if (expectedColor === 'green') {
+      // Для зеленого состояния: только is-valid, без is-invalid
+      await expect(field).toHaveClass(/is-valid/);
+      await expect(field).not.toHaveClass(/is-invalid/);
+    } else {
+      // Для красного состояния: должен быть is-invalid (может быть с is-valid одновременно)
+      await expect(field).toHaveClass(/is-invalid/);
+      // НЕ проверяем отсутствие is-valid для красного состояния
+      // так как JavaScript может добавить: "form-control is-valid is-invalid"
+    }
+  }
+
+  @logStep('Check field has validation error state: {fieldName}')
+  async checkFieldHasValidationError(fieldName: string) {
+    const field = this.getFieldByName(fieldName);
+    // Проверяем только наличие is-invalid класса
+    await expect(field).toHaveClass(/is-invalid/);
+  }
+
+  @logStep('Check field has valid state: {fieldName}')
+  async checkFieldHasValidState(fieldName: string) {
+    const field = this.getFieldByName(fieldName);
+    // Проверяем что поле имеет is-valid и НЕ имеет is-invalid
+    await expect(field).toHaveClass(/is-valid/);
+    await expect(field).not.toHaveClass(/is-invalid/);
+  }
+
+  @logStep('Check validation error message: {fieldName}')
+  async checkValidationErrorMessage(fieldName: string, expectedMessage: string) {
+    const errorElement = this.getErrorElementByFieldName(fieldName);
+    const field = this.getFieldByName(fieldName);
+
+    // Проверяем что поле имеет класс is-invalid
+    await expect(field).toHaveClass(/is-invalid/);
+
+    // Проверяем что сообщение об ошибке содержит ожидаемый текст
+    await expect(errorElement).toContainText(expectedMessage);
```

Comment:
Ошибка имеет определенный текст
```suggestion
    await expect(errorElement).toEqual(expectedMessage); // toBe
```

Reply:
принял